Makefile

```diff
@@ -154,3 +154,9 @@ $(ELVIS):
 
 elvis: $(ELVIS)
 	$(ELVIS) --config make/elvis.config rock
+
+ci: compile xref build-plt diff sup_completion build-ci-release compile-test eunit elvis
```

Comment:
I would suggest leaving CI things in Travis's file. Other CI providers have other semantics anyway (in the way they fetch the git repo for example).

I would instead prefer a `make` target that would be used as a pre-commit hook (albeit taking too long to run):
```Makefile
checks: export TO_DIALYZE = $(shell git diff --name-only master... -- applications/ core/)
checks: kazoo elvis xref dialyze sup_completion build-release
        ./scripts/validate-js.sh $(shell find applications/ core/ -name '*.json')
```

I excluded compile-test as that recompiles the whole kazoo with -DTEST, leaving the dev in a possible panic state.